### PR TITLE
[WIP] xSDK initialize libraries for the exascale computing project

### PR DESCRIPTION
**Pull request by costrouc**
###### Motivation for this change

I want to make nixos/nixpkgs the place to go for hpc. Currently many scientists with these libraries use https://spack.readthedocs.io/en/latest/ and these are the important core libraries. More to come listed in https://xsdk.info/packages/.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/science/math/dealii/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    rev = "v${version}";
+    sha256 = "01vz85fcm092bkbk4ib356j80b2hdwcxykkrhyrkw2pbas2npvnf";
+  };
+
+  buildInputs = [ cmake mpi ];
+
+  meta = with stdenv.lib; {
+    homepage = https://github.com/dealii/dealii;
+    description = "Computational solution of partial differential equations using adaptive finite elements";
+    license = licenses.lgpl3;
```
Comment (veprbl) [inline pkgs/development/libraries/science/math/dealii/default.nix]:
```suggestion
    license = licenses.lgpl21Plus;
```

File: pkgs/development/libraries/science/math/hypre/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    cd src
+    asdfasdfasdfasdf # intentionally fail build
+  '';
+
+  buildInputs = [ mpi gfortran cmake ];
+
+  meta = with stdenv.lib; {
+    homepage = https://github.com/hypre-space/hypre;
+    description = "Library for preconditioners and solvers featuring multigrid methods for the solution of large, sparse linear systems of equations";
+    license = licenses.lgpl2;
```
Comment (veprbl) [inline pkgs/development/libraries/science/math/hypre/default.nix]:
```suggestion
    license = licenses.lgpl21;
```

Comment (costrouc) [reply pkgs/development/libraries/science/math/hypre/default.nix]: Do you have a tool that you use for analyzing licenses? Or do you do this work manually?

Comment (veprbl) [reply pkgs/development/libraries/science/math/hypre/default.nix]: I usually just look at github repo, it has automatic detection that works most of the times. This time I had to actually look in the README files. So, yes, it is a manual process for me. I think debian might have some tooling to do this automatically.

File: pkgs/development/libraries/science/math/hypre/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  src = fetchFromGitHub {
+    owner = "hypre-space";
+    repo = "hypre";
+    rev = "v${version}";
+    sha256 = "1gha571flc6q513137q4416hzpzb6qn2b9hlp3swik5150m91bbb";
+  };
+
+  preConfigure = ''
+    cd src
+    asdfasdfasdfasdf # intentionally fail build
```
Comment (memberbetty) [inline pkgs/development/libraries/science/math/hypre/default.nix]:
```suggestion
    false # intentionally fail build
```

**Comment by veprbl** [comment]
@costrouc Are you still interested in working on this?

**Comment by costrouc** [comment]
Likely not. This can be closed. I'd I were to move the PR forward I'll make a PR for each separate package

**Pull request closed without merging**